File: api-net/Models/Customer.cs
namespace LiftOtonom.Api.Models;

public class Customer
{
    public long Id { get; set; }
    public long TenantId { get; set; }
    public string Type { get; set; } = "corporate";
    public string Name { get; set; } = "";
    public string? TaxNumber { get; set; }
    public string? TaxOffice { get; set; }
    public string? IdNumber { get; set; }
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public string? Address { get; set; }
    public string? District { get; set; }
    public string? City { get; set; }
    public long? RegionId { get; set; }
    public string? Notes { get; set; }
    public bool IsActive { get; set; } = true;
    public bool IsSample { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime? DeletedAt { get; set; }

    public const string TypeIndividual = "individual";
    public const string TypeCorporate = "corporate";
}

File: api/app/Models/Customer.php
<?php
// app/Models/Customer.php

namespace App\Models;

use App\Models\Concerns\HasTenant;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Customer extends Model
{
    use HasTenant, SoftDeletes;

    public const TYPE_INDIVIDUAL = 'individual';
    public const TYPE_CORPORATE  = 'corporate';

    protected $fillable = [
        'tenant_id', 'type', 'name', 'tax_number', 'tax_office', 'id_number',
        'phone', 'email', 'address', 'district', 'city', 'region_id',
        'notes', 'is_active', 'is_sample',
    ];

    protected function casts(): array
    {
        return [
            'is_active' => 'boolean',
            'is_sample' => 'boolean',
        ];
    }

    public function region(): BelongsTo
    {
        return $this->belongsTo(Region::class);
    }

    public function buildings(): HasMany
    {
        return $this->hasMany(Building::class);
    }

    public function currentAccount()
    {
        return $this->hasOne(CurrentAccount::class);
    }
}
